smart_open/smart_open_lib.py

```diff
@@ -833,12 +833,17 @@ def write(self, b):
 
         if self.chunk_bytes >= self.min_part_size:
             buff = b"".join(self.lines)
-            logger.info("uploading part #%i, %i bytes (total %.3fGB)" % (self.parts, len(buff), self.total_size / 1024.0 ** 3))
+            buff_len = len(buff)
+            logger.info("uploading part #%i, %i bytes (total %.3fGB)" % (self.parts, buff_len, self.total_size / 1024.0 ** 3))
             self.mp.upload_part_from_file(BytesIO(buff), part_num=self.parts + 1)
             logger.debug("upload of part #%i finished" % self.parts)
             self.parts += 1
             self.lines, self.chunk_bytes = [], 0
 
+            return buff_len
+
+        return 0
```

Comment:
Is this correct? What is the actual contract for this number -- the number of bytes accepted (then this shouldn't be 0), or the number of bytes passed to lower I/O layer (then 0 OK), or number of bytes actually written/ACK'd/synced to some physical medium...?